### ISSUE TITLE
Resolvers unable to receive arguments when inside of a federation reference resolver.

### ISSUE THREAD

**Issue by prowe** [graphql-dotnet/graphql-dotnet#4086]
## Description

Unable to read arguments to resolvers on fields of entities that are resolved via federation `__entities` selection. 

### Steps to reproduce

Create a federation 2 schema first project, Add a reference resolver and then add a field on that type that has an argument. For example:

```graphql

type Droid @key(fields: "id") {
  id: ID!
  greet(name: String!): String
}
```

if the following query is run,
```graphql
query {
  __entities(representations: [{ __typename: "Droid", id: "123" }]) {
    ... on Droid {
      greet(name: "smith")
    }
  }
}
```

then the resolver for `greet` will never get the `name` argument of "smith"

### Expected result

The resolver should get the value of the arguments.

### Actual result

adding a C# argument to the resolver method always results in either `null` or a default if one is specified, but never the one in the query.

IFieldResolveContext.GetArgument also returns null

### Environment
GraphQL.Server.All=8.0.3
GraphQL.SystemTextJson=8.2.0
Sdk="Microsoft.NET.Sdk.Web"
.NET 8.0

**Comment by Shane32** [comment]
This bug was introduced with the new argument parsing code in v8.  The patch will be released shortly in version 8.2.1

**Comment by Shane32** [comment]
8.2.1 has been released and is available on NuGet with the patch
